src/Parser/PhpDocParser.php

```diff
@@ -340,6 +342,16 @@ private function parseMethodTagValue(TokenIterator $tokens): Ast\PhpDoc\MethodTa
 			$returnType = $isStatic ? new Ast\Type\IdentifierTypeNode('static') : null;
 			$methodName = $returnTypeOrMethodName->name;
 			$isStatic = false;
+		} elseif (
+			$returnTypeOrMethodName instanceof Ast\Type\IntersectionTypeNode
```

Comment:
No, that's a horrible hack 😂

Reply:
I'm sorry for this 😊

Reply:
How to solve this in a better way? I think this would require a change in the type parser as it is now to greedy.